### PR TITLE
issue-1

### LINKED ISSUE #1: Modify generators to use GrammarSpec
Instead of using the Grammar directly, all generators should use the `GrammarSpec`.

### PR DESCRIPTION
**Pull request by aeram0**
Closes #1
Modifies generator to use GrammarSpec, in favor of Grammar

### CODE REVIEW

File: kanonic-runtime/src/main/kotlin/io/johnedquinn/kanonic/machine/Action.kt [hunk 1]
```diff
@@ -5,6 +5,6 @@ sealed class Action {
         is ShiftAction -> "S${this.state}"
         is ReduceAction -> "R${this.rule}"
         is AcceptAction -> "ACCEPT"
-        else -> error("None found")
+//        else -> error("None found")
```
Comment (johnedquinn) [inline kanonic-runtime/src/main/kotlin/io/johnedquinn/kanonic/machine/Action.kt:8]: nit

**Comment by johnedquinn** [comment]
Absolutely great first PR!

Left small nits, but don't need to address. When you merge, can you Squash and merge and rename to something like "Removes use of Grammar in gen" or "Modifies generator to use GrammarSpec, in favor of Grammar"

**Review by johnedquinn — APPROVED** [review]

File: kanonic-runtime/src/main/kotlin/io/johnedquinn/kanonic/machine/Action.kt (hunk 1 above)
Comment (aeram0) [reply kanonic-runtime/src/main/kotlin/io/johnedquinn/kanonic/machine/Action.kt]: resolved

File: kanonic-gen/src/main/kotlin/io/johnedquinn/kanonic/gen/RuleSpec.kt [hunk 2]
```diff
@@ -8,5 +8,6 @@ class RuleSpec(
     val visitMethodName: String,
     val variants: List<VariantSpec>,
     val className: ClassName,
-    val generated: Boolean
+    val generated: Boolean,
+    var alias: String = "None"
```
[the remark this answers is not in the document]
Comment (aeram0) [reply kanonic-gen/src/main/kotlin/io/johnedquinn/kanonic/gen/RuleSpec.kt]: resolved by e0bdf7ae23b9db5567c1f42f9ebd002b0f22fd2b

**Pull request merged**